llama_index/llm_predictor/base.py

```diff
@@ -169,22 +229,26 @@ def stream(self, prompt: BasePromptTemplate, **prompt_args: Any) -> TokenGen:
             stream_tokens = stream_completion_response_to_tokens(stream_response)
         return stream_tokens
 
-    async def apredict(self, prompt: BasePromptTemplate, **prompt_args: Any) -> str:
+    async def apredict(
+        self,
+        prompt: BasePromptTemplate,
+        output_cls: Optional[BaseModel] = None,
+        **prompt_args: Any
+    ) -> str:
         """Async predict."""
         self._log_template_data(prompt, **prompt_args)
 
         if self._llm.metadata.is_chat_model:
-            messages = prompt.format_messages(llm=self._llm, **prompt_args)
-            messages = self._extend_messages(messages)
-            chat_response = await self._llm.achat(messages)
-            output = chat_response.message.content or ""
-            # NOTE: this is an approximation, only for token counting
-            formatted_prompt = messages_to_prompt(messages)
+            output = self._handle_predict_chat_model(
+                prompt=prompt,
+                chat_fn=self._llm.achat,
```

Comment:
I think this will actually break for async

Since achat is never awaited, `handle_predict_chat_model` will likely error out 

Programs also have an `acall()` that should be used for async as well 🤔 

Hmm, will sleep on this one

Reply:
Might have to revert to something closer to the original LLMPredictor -- but with some helper function to get the program?